File: src/boil/detectors/cannot_open_file.go
// Detector for "Cannot open file" errors.
package detectors

// CannotOpenFileDetector detects "Cannot open file" errors
type CannotOpenFileDetector struct {
	*BaseDetector
}

func NewCannotOpenFileDetector() (*CannotOpenFileDetector, error) {
	base, err := NewBaseDetector(
		"CannotOpenFileDetector",
		100,
		map[string]string{
			"missing_file": `[Ee]rror:?\s+Cannot open file\s+['"](?P<file_path>[^'"]+)['"]`,
		},
		[]DetectorExample{
			{
				Name:     "cannot_open_single_quotes",
				Input:    "Error: Cannot open file 'config.json'",
				ClueType: "missing_file",
				Context:  map[string]string{"file_path": "config.json"},
			},
			{
				Name:     "cannot_open_double_quotes",
				Input:    `error Cannot open file "main.py"`,
				ClueType: "missing_file",
				Context:  map[string]string{"file_path": "main.py"},
			},
		},
	)
	if err != nil {
		return nil, err
	}
	return &CannotOpenFileDetector{base}, nil
}

File: pipeline/detectors/cannot_open_file.py
"""
Detector for generic "Cannot open file" errors.
"""

import re
import typing as T
from pipeline.detectors.base import Detector
from pipeline.models import ErrorClue


class CannotOpenFileDetector(Detector):
    """
    Detect generic "Cannot open file" errors from programs.

    Matches patterns like:
    - Error: Cannot open file 'example.c'
    - Error: Cannot open file "example.py"
    - error: Cannot open file 'test.txt'
    """

    PATTERNS = {
        "missing_file": r"[Ee]rror:?\s+Cannot open file\s+['\"](?P<file_path>[^'\"]+)['\"]",
    }

    EXAMPLES = [
        (
            "Error: Cannot open file 'example.c'",
            {
                "clue_type": "missing_file",
                "confidence": 1.0,
                "context": {"file_path": "example.c"},
            },
        ),
        (
            "error: Cannot open file \"example.py\"",
            {
                "clue_type": "missing_file",
                "confidence": 1.0,
                "context": {"file_path": "example.py"},
            },
        ),
    ]
